### ISSUE TITLE
Multiple Resends

### ISSUE THREAD

**Issue by scmurray1** [geeknam/python-gcm#98]
I'm having an issue where I send one notification but am receiving 5 of them on my device.

**Comment by ishansharma29** [comment]
There is some bug in this API. The default retry strategy retries sending notification for 5 times in case of any errors. And I don't know exactly what but they are catching some exception that isn't really a valid exception and retrying on that. To overcome this, I'd manually set retry to 1 so that I only get one notification. geeknam@ please work on fixing this please.

**Comment by mnemonicflow** [comment]
@scmurray1 @ishansharma29 Will look into this. Thanks for reporting!

**Comment by mnemonicflow** [comment]
@scmurray1 
1) Can you provide code details and input data on how exactly are you calling the API, I cannot reproduce the problem even after explicitly rasing GCMUnavailableException and returning error from make_request a certain number of attempts.
2) Can you please enable logging by adding gcm = GCM(API_KEY, debug=True) to the object and try to send a notification that raises the GCMUnavailableException and attach the logs here?
3) Are you using json_request() or plaintext_request() ?

@ishansharma29 What do you mean by an exception that is not really a valid exception?
An exception that is not coming from the GCM server but from the code itself? i.e. a situation where an HTTP POST request is actually made and a success response is returned, but an error is raised by some other part of the code and it retries to send the notification again. 
For example, in the case of json_request() GCMUnavailableException is only raised in two places: [#L314](https://github.com/geeknam/python-gcm/blob/master/gcm/gcm.py)
```python
       elif response.status_code == 503:
            raise GCMUnavailableException("GCM service is unavailable")
        else:
            error = "GCM service error: %d" % response.status_code
            raise GCMUnavailableException(error)
```
Maybe GCM server actually sends the notification and returns an HTTP response status code other than 200 for success?

**Comment by scmurray1** [comment]
I'm using plaintext from the example provided. I'm using python 2.9. I'll enable debugging and look at the output when I can. Probably a little later today. Thanks for the reply.

from gcm import GCM

gcm = GCM(API_KEY)
data = {'param1': 'value1', 'param2': 'value2'}

# Plaintext request
reg_id = '12'
gcm.plaintext_request(registration_id=reg_id, data=data)

**Comment by scmurray1** [comment]
Ok here is the ouput. I had changed the retries from 5 to 1 but changed them back to 5.

[gcmdebug.txt](https://github.com/geeknam/python-gcm/files/77471/gcmdebug.txt)

**Comment by mnemonicflow** [comment]
@scmurray1 @ishansharma29 The problem has been fixed in [develop branch](https://github.com/geeknam/python-gcm/tree/develop) https://github.com/geeknam/python-gcm/commit/6ef1313455f049f318c1381cb5c37238ed24a1aa#diff-464232c9a5921970b623c217c57caae7R443 , after the major refactoring that took place last year seems like an error slipped through. 

Sorry for the incovenience, but the recommended way to use GCM is with the JSON request payload, that's why the plaintext_request was a bit neglected.

Anyway, I've updated the README to reflect this and also added an examples directory https://github.com/geeknam/python-gcm/tree/develop/examples

To solve this problem try the fix introduced in the [6ef1313455f049f318c1381cb5c37238ed24a1aa](https://github.com/geeknam/python-gcm/commit/6ef1313455f049f318c1381cb5c37238ed24a1aa) commit on the develop branch

**Comment by scmurray1** [comment]
Works perfect. Thank you for the quick fix.
